MonoGame.Framework/Graphics/GraphicsDevice.cs

```diff
@@ -890,6 +893,54 @@ public void SetVertexBuffer(VertexBuffer vertexBuffer, int vertexOffset)
                                    ? _vertexBuffers.Clear()
                                    : _vertexBuffers.Set(vertexBuffer, vertexOffset);
         }
+
+        private void _SetVertexBuffers(VertexBufferBinding[] vertexBuffers, int bindingCount)
+        {
+            if (vertexBuffers == null || vertexBuffers.Length == 0 || bindingCount == 0)

```

Comment:
Do you really need this path?  Only your overloaded `SetVertexBuffers` call this and it always passes a buffer and always has a count and length.